### PR TITLE
Fsm opposite transfer funcs

### LINKED ISSUE #742: Clean up FSM alarm behaviour
### Summary
FSM alarm behavior is hacked in right now since new PAPPS has no alarm output. We should probably fix this.

### PR DESCRIPTION
**Pull request by gtaharaedmonds**
### Summary
As per T.4.2.3 the 2 apps sensors may need opposite transfer functions. Not sure if this is required in our situation, but it is now implemented to be safe. Conducted some basic testing on the car, but unit tests aren't fixed yet. 

### Changelist 
1. Secondary apps counts down instead of up, papps still counts up.
2. Fixed papps alarm detection.
3. Configured min/max values with testing on the car.

### Testing Done
Basic testing on the car, still need to fix unit tests.

### Resolved Issues
Fixes #742, #737

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: boards/FSM/Test/Src/Test_AcceleratorPedal.cpp [hunk 1]
```diff
@@ -6,10 +6,10 @@ extern "C"
 }
 
 FAKE_VALUE_FUNC(bool, is_primary_encoder_alarm_active);
-FAKE_VOID_FUNC(reset_primary_encoder_counter);
+FAKE_VOID_FUNC(set_primary_encoder_counter, uint32_t);
 FAKE_VALUE_FUNC(uint32_t, get_primary_encoder_counter);
 FAKE_VALUE_FUNC(bool, is_secondary_encoder_alarm_active);
-FAKE_VOID_FUNC(reset_secondary_encoder_counter);
+FAKE_VOID_FUNC(set_secondary_encoder_counter, uint32_t); //is this syntax correct?
```
Comment (pasitlaothamatas) [inline boards/FSM/Test/Src/Test_AcceleratorPedal.cpp:12]: should be

**Comment by timbitss** [comment]
Has my latest commit on the main branch been merged into your fsm_opposite_transfer_fcns branch? I think there are differences only in the `App_CanMsgs.dbc` file which or may not interfere with the inverters.

File: boards/FSM/Src/Io/Io_SecondaryScancon2RMHF.c [hunk 2]
```diff
@@ -22,7 +22,7 @@ uint32_t Io_SecondaryScancon2RMHF_GetEncoderCounter(void)
     return __HAL_TIM_GET_COUNTER(secondary_encoder_htim);
 }
 
-void Io_SecondaryScancon2RMHF_ResetEncoderCounter(void)
+void Io_SecondaryScancon2RMHF_SetEncoderCounter(uint32_t counter_value)
```
Comment (pasitlaothamatas) [inline boards/FSM/Src/Io/Io_SecondaryScancon2RMHF.c:25]: may be better to pass in a uint16_t?

File: boards/FSM/Src/main.c [hunk 3]
```diff
@@ -959,7 +965,7 @@ void RunTask1kHz(void const *argument)
         const uint32_t current_time_ms = osKernelSysTick() * portTICK_PERIOD_MS;
 
         App_SharedClock_SetCurrentTimeInMilliseconds(clock, current_time_ms);
-        App_FsmWorld_UpdateSignals(world, current_time_ms);
+        //        App_FsmWorld_UpdateSignals(world, current_time_ms);
```
Comment (pasitlaothamatas) [inline boards/FSM/Src/main.c:968]: may want to fix this

**Comment by pasitlaothamatas** [comment]
I think it'd be nice to have count up and count down be the same function. I notice a lot of repeated code in these two fxns

File: boards/FSM/Inc/App/App_AcceleratorPedals.h [hunk 4]
```diff
@@ -2,6 +2,23 @@
 
 #include <stdbool.h>
 
+// Papps counts up
+#define PAPPS_ENCODER_FULLY_PRESSED_VALUE (1000U)
+#define PAPPS_ENCODER_UNPRESSED_VALUE (1353U)
```
Comment (pasitlaothamatas) [inline boards/FSM/Inc/App/App_AcceleratorPedals.h:7]: fully pressed value is smaller than unpressed vale for PAPPS?

**Pull request closed without merging**